test: /analytics/get_quest_participation?id=${id}

EjembiEmmanuel opened the pull request:
# Pull Request type

Test

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

Resolves: #622

EjembiEmmanuel committed:
test: /analytics/get_quest_participation?id=${id}

ayushtom commented on tests/services/getQuestParticipant.test.js:
```diff
@@ -0,0 +1,68 @@
+    expect(result).toBeNull();
+  });
+
+  it("should handle no response", async () => {
```
this returns a valid response, can you handle this as a error 500 when the api fails to get data?

ayushtom commented on tests/services/getQuestParticipant.test.js:
```diff
@@ -0,0 +1,68 @@
+  });
+
+  it("should handle fetch errors gracefully", async () => {
+    const mockResponse = "Quest not found";
```
not a valid response

ayushtom replied:
you should be getting a empty array here ideally

ayushtom commented on tests/services/getQuestParticipant.test.js:
```diff
@@ -0,0 +1,68 @@
+  });
+
+  it("should fetch and return data for a valid id", async () => {
+    const mockData = {
```
i would suggest using the api once to see what is the ideal response you get from the api for this and also add that as a return type in the apiService.ts

EjembiEmmanuel committed:
refac(getQuestParticipant): add return type to
getQuestParticipant()
test(getQuestParticipant): validate empty response
test(getQuestParticipant): validate unexpected response

ayushtom commented:
hey i would also need you to add the test cases to apiServices.test.js

ayushtom commented:
test cases look good to me

EjembiEmmanuel committed:
test(getQuestParticpation): validate
getQuestParticpation() API calls

EjembiEmmanuel commented:
@ayushtom I've added the test cases to apiServices.test.js

EjembiEmmanuel committed:
refac(getQuestParticpantion): fix url typo

EjembiEmmanuel committed:
test(getQuestParticipantion): getQuestParticipation

- fix wrong function call
- run prettier

ayushtom commented on services/apiService.ts:
```diff
@@ -293,7 +293,9 @@ export const getQuestActivityData = async (id: number) => {
-export const getQuestsParticipation = async (id: number) => {
+export const getQuestsParticipation = async (
+  id: number
+): Promise<object[] | undefined> => {
```
you need to add the type in backTypes.ts like - 
type QuestParticipants = {
  name: String;
  desc: String;
participants: String;
}[];

and import that type here as Promise<QuestParticipants>

EjembiEmmanuel replied:
Alright

ayushtom commented:
Also you can remove the `tests/services/getQuestParticipant.test.js` from changelog

EjembiEmmanuel committed:
refac: add type declaration

- declare type QuestParticipation in backTypes.d.ts
- remove getQuestParticipation.test.ts from change log

EjembiEmmanuel commented:
@ayushtom I've effected the changes

ayushtom committed:
Merge branch 'testnet' into test/getQuestParticpant

ayushtom commented:
@EjembiEmmanuel your build is failing

EjembiEmmanuel commented:
@ayushtom please can you specify why it is failing

EjembiEmmanuel committed:
fix: fix QuestParticipation type issues

EjembiEmmanuel committed:
Merge branch 'test/getQuestParticpant' of github.com:EjembiEmmanuel/starknet.quest into test/getQuestParticpant

EjembiEmmanuel committed:
chore: resolve conflicts

EjembiEmmanuel committed:
fix: fix linting issue

EjembiEmmanuel commented:
@ayushtom I think I've fixed it. The issue was caused by the QuestParticipation type I declared in backTypes.d.ts. I had to refactor the usage in starknet.quest/app/analytics/[questId]/page.tsx to be compatible with it.

ayushtom committed:
Merge branch 'testnet' into test/getQuestParticpant

ayushtom committed:
feat: add types in state

ayushtom commented:
there was a build issue. I resolve it in my commit

EjembiEmmanuel commented:
Okay

ayushtom committed:
feat: add correct params

ayushtom reviewed (approved):
lgtm
